schema/v5.0/CVE_JSON_5.0.schema

```diff
@@ -198,6 +175,62 @@
                         "https://wordpress.org/plugins"
                     ]
                 },
+                "packageName": {
+                    "type": "string",
+                    "description": "Name or identifier of the affected software package as used in the package collection.",
+                    "minLength": 1,
+                    "maxLength": 2058
+                 },
+                "cpes": {
+                    "type": "array",
+                    "description": "Affected products defined by CPE. This is an array of CPE values (vulnerable and not), we use an array so that we can make multiple statements about the same version and they are separate (if we used a JSON object we'd essentially be keying on the CPE name and they would have to overlap). Also this allows things like cveDataVersion or cveDescription to be applied directly to the product entry. This also allows more complex statements such as \"Product X between versions 10.2 and 10.8\" to be put in a machine-readable format. As well since multiple statements can be used multiple branches of the same product can be defined here.",
+                    "uniqueItems": true,
+                    "items": {
+                        "type": "object",
+                        "properties": {},
+                        "minProperties": 1
+                    }
```

Comment:
Better to define it:

```
"items": {
                        "title": "CPE Name",
                        "type": "string",
                        "description":"Common Platform Enumeration (CPE) Name in either 2.2 or 2.3 format",
                        "pattern": "([c][pP][eE]:\/[AHOaho]?(:[A-Za-z0-9\._\-~%]*){0,6}|cpe:2\.3:[aho\*\-](:(((\?*|\*?)([a-zA-Z0-9\-\._]|(\\[\\\*\?!"#$$%&'\(\)\+,/:;<=>@\[\]\^`\{\|}~]))+(\?*|\*?))|[\*\-])){5}(:(([a-zA-Z]{2,3}(-([a-zA-Z]{2}|[0-9]{3}))?)|[\*\-]))(:(((\?*|\*?)([a-zA-Z0-9\-\._]|(\\[\\\*\?!"#$$%&'\(\)\+,/:;<=>@\[\]\^`\{\|}~]))+(\?*|\*?))|[\*\-])){4})"
                        "minLength": 1,
                        "maxLength": 2000
                    }
```

Reply:
Done. The regexp as written didn't work because it wasn't properly escaped for use in a JSON string literal. I dug up the original (two) at http://cybox.mitre.org/language/version2.0.1/xsddocs/extensions/platform/cpe2.3/1.0.1/cpe-naming_2_3_xsd.html and prepared a new JSON string literal.